internal/cmd/stack/run_cancel.go

```diff
@@ -0,0 +1,56 @@
+package stack
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/shurcooL/graphql"
+	"github.com/urfave/cli/v2"
+
+	"github.com/spacelift-io/spacectl/internal/cmd/authenticated"
+)
+
+func runCancel() cli.ActionFunc {
+	return func(cliCtx *cli.Context) error {
+		stackID, err := getStackID(cliCtx)
+		if err != nil {
+			return err
+		}
+
+		var mutation struct {
+			RunDiscard struct {
+				ID string `graphql:"id"`
+			} `graphql:"runCancel(stack: $stack, run: $run)"`
+		}
+
+		variables := map[string]interface{}{
+			"stack": graphql.ID(stackID),
+			"run":   graphql.ID(cliCtx.String(flagRequiredRun.Name)),
+		}
+
+		ctx := context.Background()
+
+		if err := authenticated.Client.Mutate(ctx, &mutation, variables); err != nil {
+			return err
+		}
+
+		fmt.Println("You have successfully canceled a run")
```

Comment:
Tiny tiny minor nitpick: 
```suggestion
		fmt.Println("You have successfully canceled the run")
```